tools/data_integrity/update_candidate_latest_diagnosis.php

```diff
@@ -0,0 +1,129 @@
+    WHERE Entity_type='Human' AND Active='Y'",
+    []
+);
+
+// get configured diagnosis trajectories in DESC order by project
+$diagnosisTrajectories = $DB->pselect(
+    "SELECT * FROM diagnosis_evolution
+    ORDER BY orderNumber DESC",
+    []
+);
+
+// if no diagnosis trajectories are defined, return null
+if (is_null($diagnosisTrajectories)) {
+    echo "There are no configured Diagnosis Trajectories. Nothing to update.\n";
+    exit;
+}
+
+// Format trajectories by project
+$diagnosisTrajectoryByProject = [];
+foreach ($diagnosisTrajectories as $key => $data) {
+    $diagnosisTrajectoryByProject[$data['ProjectID']][] = $data;
+}
+
+$loris = new \LORIS\LorisInstance(
+    \NDB_Factory::singleton()->database(),
+    \NDB_Factory::singleton()->config(),
+    [
+        "project/modules",
+        "modules",
+    ]
+);
+
+foreach ($candIDs as $k => $candID) {
+    $candidate       = \Candidate::singleton(new CandID($candID));
+    $candidateVisits = $candidate->getListOfVisitLabels();
+
+    foreach ($diagnosisTrajectoryByProject as $projectID => $projectTrajectories) {
+        foreach ($projectTrajectories as $key => $data) {
+            // search if candidate has a matching visit
+            $sessionID = array_search(
+                $data['visitLabel'],
+                array_reverse($candidateVisits, true)
+            );
+            if ($sessionID) {
+                $matchingVL = $candidateVisits[$sessionID];
+
+                // Find instance of instrument
+                $commentID = $DB->pselectOne(
+                    "SELECT CommentID FROM flag f
+                    WHERE f.SessionID=:sid
+                    AND f.Test_name=:tn
+                    AND CommentID NOT LIKE 'DDE%'",
+                    [
+                        'sid' => $sessionID,
+                        'tn'  => $data['instrumentName'],
+                    ]
+                );
+
+                // If instrument does not exist, go to next diagnosis track
+                if (!$commentID) {
+                    continue;
+                }
+
+                // get instrument instance data
+                $instrument     = \NDB_BVL_Instrument::factory(
+                    $loris,
+                    $data['instrumentName'],
+                    $commentID
+                );
+                $instrumentData = $instrument->getInstanceData();
+
+
+                $diagnosis    = [];
+                $sourceFields = explode(",", $data['sourceField']);
+                foreach ($sourceFields as $k => $fieldName) {
+                    if (!isset($instrumentData[$fieldName])) {
+                        continue;
+                    }
+                    $diagnosis[$fieldName] = $instrumentData[$fieldName];
+                }
+
+                if (!empty($diagnosis)) {
+                    $confirmed = \TimePoint::singleton(
+                        new SessionID($sessionID)
+                    )->getVisitStatus() === 'Pass' ?
```

Comment:
similarly to above comment, should the diagnosis be confirmed on the visit status passing (as here), the approval status starting (as via control panel), or the approval status passing? on the jsx, a confirmed diagnosis is described as "one that belongs to an approved visit" but im not quite sure what this means.

Reply:
Yes this was recently changed after discussion at LORIS meeting. Visit pass is sufficient.

Reply:
how come visit pass isn't sufficient when using the control panel, where the timepoint needs to be in an approval stage i.e. when sent to DCC? i don't see why the workflow would be different between front-end and back-end data entry. whats the reasoning?